### DISCUSSION TITLE
My i8042prt is fucked up and causes high latency when only the caps lock key is pressed.

### DISCUSSION THREAD

**Discussion by Zylon676** [seerge/g-helper#2873] [Q&A]
Could anyone please help check the following parameters in the registry relating to i8042prt, as this the only thing i found that helps offset this, I understand this not right place to ask this but i have asked everyone and no-one has been a solution in any capacity and this the most active code based asus related thread.

I would appreciate any help at this point.
![Screenshot 2024-07-23 125815](https://github.com/user-attachments/assets/24aae5a2-da8a-40ee-a86d-76eed211a7f2)
<img width="651" alt="Screenshot 2024-07-22 175506" src="https://github.com/user-attachments/assets/48b32ea6-d663-41e6-85bb-9fe27efb58ee">
![Screenshot 2024-07-22 175515](https://github.com/user-attachments/assets/263316ea-f81a-4c8b-8a18-01b9e34c0769)

![Screenshot 2024-07-22 175522](https://github.com/user-attachments/assets/272cf5be-7c1a-4392-964a-ae826454551f)
_________________________________________________________________________________________________________
CONCLUSION
_________________________________________________________________________________________________________
Your system appears to be having trouble handling real-time audio and other tasks. You are likely to experience buffer underruns appearing as drop outs, clicks or pops. One or more ISR routines that belong to 
a driver running in your system appear to be executing for too long. One problem may be related to power management, disable CPU throttling settings in Control Panel and BIOS setup. Check for BIOS updates. 
LatencyMon has been analyzing your system for  0:01:14  (h:mm:ss) on all processors.

_________________________________________________________________________________________________________
SYSTEM INFORMATION
_________________________________________________________________________________________________________
Computer name:                                        PVPC
OS version:                                           Windows 11, 10.0, version 2009, build: 22621 (x64)
Hardware:                                             ROG Zephyrus G15 GA503RW_GA503RW, ASUSTeK COMPUTER INC.
BIOS:                                                 GA503RW.317
CPU:                                                  AuthenticAMD AMD Ryzen 9 6900HS with Radeon Graphics 
Logical processors:                                   16
Processor groups:                                     1
Processor group size:                                 16
RAM:                                                  15608 MB total

_________________________________________________________________________________________________________
CPU SPEED
_________________________________________________________________________________________________________
Reported CPU speed (WMI):                             3301 MHz
Reported CPU speed (registry):                        3294 MHz

Note: reported execution times may be calculated based on a fixed reported CPU speed. Disable variable speed settings like Intel Speed Step and AMD Cool N Quiet in the BIOS setup for more accurate results.

_________________________________________________________________________________________________________
MEASURED INTERRUPT TO DPC LATENCIES
_________________________________________________________________________________________________________
The interrupt to DPC latency reflects the measured interval in which a DPC could execute in response to a hardware request from the moment the interrupt service routine started execution.

Highest measured interrupt to DPC latency (µs):       79645.40
Average measured interrupt to DPC latency (µs):       17.318099

_________________________________________________________________________________________________________
 REPORTED ISRs
_________________________________________________________________________________________________________
Interrupt service routines are routines installed by the OS and device drivers that execute in response to a hardware interrupt signal.

Highest ISR routine execution time (µs):              81242.313297
Driver with highest ISR routine execution time:       i8042prt.sys - i8042 Port Driver, Microsoft Corporation

Highest reported total ISR routine time (%):          0.006850
Driver with highest ISR total time:                   i8042prt.sys - i8042 Port Driver, Microsoft Corporation

Total time spent in ISRs (%)                          0.009239

ISR count (execution time <250 µs):                   6504
ISR count (execution time 250-500 µs):                0
ISR count (execution time 500-1000 µs):               0
ISR count (execution time 1000-2000 µs):              0
ISR count (execution time 2000-4000 µs):              0
ISR count (execution time >=4000 µs):                 0

_________________________________________________________________________________________________________
REPORTED DPCs
_________________________________________________________________________________________________________
DPC routines are part of the interrupt servicing dispatch mechanism and disable the possibility for a process to utilize the CPU while it is interrupted until the DPC has finished execution.

Highest DPC routine execution time (µs):              338.124772
Driver with highest DPC routine execution time:       ntoskrnl.exe - NT Kernel & System, Microsoft Corporation

Highest reported total DPC routine time (%):          0.004616
Driver with highest DPC total execution time:         dxgkrnl.sys - DirectX Graphics Kernel, Microsoft Corporation

Total time spent in DPCs (%)                          0.024015

DPC count (execution time <250 µs):                   42158
DPC count (execution time 250-500 µs):                0
DPC count (execution time 500-10000 µs):              5
DPC count (execution time 1000-2000 µs):              0
DPC count (execution time 2000-4000 µs):              0
DPC count (execution time >=4000 µs):                 0

**Comment by spatialzebra** [comment]
Same laptop same issue
I desperately need a fix!

**Reply by Zylon676** [reply]
Yeah so I geniuely never found a fix to this, like i just don't use the caps lock key as much, but my suggestion to you is when typing using the key press it once wait 5 sec then turn it off. I have feel it may be related to the layer drivers but I just didn't test that, so if you want export the registry and and then delete the layer drivers, restart, and see if that fixs the issue, if not then add the PollStatusIterations [REG_DWORD] into the parameters section, however it does cause a touch of input lag, but I didn't test it for that long either.

**Comment by spatialzebra** [comment]
What if I told you I did find a fix? Btw for me that latency could spike even without me pressing caps lock.

So the fix is as follows:
1. open regedit
2. go to HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\i8042prt\Parameters
3. add a DWORD value and name it PollStatusIterations
4. set to 1 or 2 (I set to 1 and saw no difference in keyboard reaction time)
5. reboot and there you go, latency is just gone!

**Reply by Zylon676** [reply]
I know that solution I was the one that got it in the first place.

**Reply by spatialzebra** [reply]
then what's wrong?
